File: 199. Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        self.res = []
        self.helper(root, 0)
        return self.res
        
    def helper(self, root, level):
        if not root:
            return
        if len(self.res) == level:
            self.res += [root.val]
        self.helper(root.right, level+1)
        self.helper(root.left, level+1)

File: 199. Binary Tree Right Side View.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    let res = [];
    var helper = function(root, level)
    {
        if (root === null)
            return;
        if (res.length == level)
            res.push(root.val);
        helper(root.right, level+1);
        helper(root.left, level+1);
    };
    helper(root, 0);
    return res;
};
